src/detect-replace.c

```diff
@@ -276,15 +276,11 @@ int DetectReplaceLongPatternMatchTest(uint8_t *raw_eth_pkt, uint16_t pktsize,
     DecodeEthernet(&th_v, &dtv, p, GET_PKT_DATA(p), pktsize);
 
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx);
     de_ctx->flags |= DE_QUIET;
 
     de_ctx->sig_list = SigInit(de_ctx, sig);
-    if (de_ctx->sig_list == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(DetectEngineAppendSig(de_ctx, sig));
     de_ctx->sig_list->next = NULL;
 
     if (de_ctx->sig_list->sm_lists_tail[DETECT_SM_LIST_PMATCH]->type == DETECT_CONTENT) {
```

Comment:
this if is an example of what I asked earlier.

Reply:
I'm rethinking what I had asked, sorry again about that. Let's leave this function untouched. There are many places where the if statements are not meant to mean to be an instant fail, but rather sort of debug what is happening along the way.

I have still answered your other questions, so you know what would be the approach, if you were to proceed with this patch :)